### PR TITLE
Extend global settings for resource quota

### LINKED ISSUE #6370: Extend global settings for resource quota

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: Extend global settings for resource quota

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6370

```release-note
Extend global settings for resource quota
```

### CODE REVIEW

File: pkg/crd/kubermatic/v1/settings.go [hunk 1]
```diff
@@ -73,6 +75,19 @@ type CleanupOptions struct {
 	Enforced bool
 }
 
+type MachineDeploymentVMResourceQuota struct {
```
Comment (moadqassem) [inline pkg/crd/kubermatic/v1/settings.go:78]: Shouldn't this follow the same resources handling and types as k8s?

Comment (zreigz) [reply pkg/crd/kubermatic/v1/settings.go:78]: no, we are going to use common parameters for all providers such as a number of vCPU and Memory. This is what we have from the provider clients and will be easy to use to filter out VMs

Comment (moadqassem) [reply pkg/crd/kubermatic/v1/settings.go:78]: Alright, I mean as long as the user can actually override those then I think we are on the good side here.

File: pkg/crd/kubermatic/v1/settings.go [hunk 2]
```diff
@@ -56,6 +56,8 @@ type SettingSpec struct {
 	RestrictProjectCreation     bool           `json:"restrictProjectCreation"`
 	EnableExternalClusterImport bool           `json:"enableExternalClusterImport"`
 
+	MachineDeploymentVMResourceQuota MachineDeploymentVMResourceQuota `json:"clusterVMResourceQuota"`
```
Comment (moadqassem) [inline pkg/crd/kubermatic/v1/settings.go]: json tag is different than the field name, is that done on purpose?

Comment (zreigz) [reply pkg/crd/kubermatic/v1/settings.go]: mistake, thanks

Comment (zreigz) [reply pkg/crd/kubermatic/v1/settings.go]: done

**Comment by zreigz** [comment]
@moadqassem PTAL

**Pull request merged**